Source/WebCore/dom/Element.cpp

```diff
@@ -2342,6 +2342,16 @@ URL Element::absoluteLinkURL() const
     return document().completeURL(linkAttribute);
 }
 
+void Element::setIsLink(bool flag)
+{
+    Style::PseudoClassChangeInvalidation styleInvalidation(*this, {
```

Comment:
Now that we are doing invalidation here, I suggest an early return for the case where the flag is not changing to avoid unnecessary work.

Reply:
Will add one!